tests/tests_sound_composer/test_sound_composer_source_harmonics.py

```diff
@@ -0,0 +1,487 @@
+EXP_ORDER_LEVEL03_XML = 5.632353957971172e-19
+EXP_STR_NOT_SET = "Harmonics source: Not set\nSource control: Not set"
+EXP_STR_ALL_SET = (
+    "Harmonics source: ''\n"
+    "\tNumber of orders: 5\n"
+    "\t\t[12. 24. 36. 48. 60.]\n"
+    "\tControl parameter: RPM, 500.0 - 3500.0 rpm\n"
+    "\t\t[500. 510. 520. 530. 540. ... 3460. 3470. 3480. 3490. 3500.]"
+    "\nSource control: \n"
+    "\tMin: 750.0\n"
+    "\tMax: 3500.0\n"
+    "\tDuration: 3.0 s"
+)
+EXP_STR_ALL_SET_10_40_VALUES = (
+    "Harmonics source: ''\n"
+    "\tNumber of orders: 40\n"
+    "\t\t[ 1.  2.  3.  4.  5.  6.  7.  8.  9. 10. 11. 12. 13. 14. 15. ... "
+    "26. 27. 28. 29. 30. 31. 32. 33. 34. 35. 36. 37. 38. 39. 40.]\n"
+    "\tControl parameter: RPM, 500.0 - 590.0 rpm\n"
+    "\t\t[500. 510. 520. 530. 540. 550. 560. 570. 580. 590.]"
+    "\nSource control: \n"
+    "\tMin: 750.0\n"
+    "\tMax: 3500.0\n"
+    "\tDuration: 3.0 s"
+)
+
+
+def test_source_harmonics_instantiation_no_arg(dpf_sound_test_server):
+    """Test SourceHarmonics instantiation without arguments."""
+    source_harmonics_obj = SourceHarmonics()
+    assert isinstance(source_harmonics_obj, SourceHarmonics)
+    assert source_harmonics_obj.source_harmonics is None
+
+
+def test_source_harmonics_instantiation_file_arg(dpf_sound_test_server):
+    """Test SourceHarmonics instantiation with file argument."""
+    source_harmonics_obj = SourceHarmonics(
+        file=pytest.data_path_sound_composer_harmonics_source_in_container
+    )
+    assert isinstance(source_harmonics_obj, SourceHarmonics)
+    assert source_harmonics_obj.source_harmonics is not None
+
+
+def test_source_harmonics___str___not_set(dpf_sound_test_server):
+    """Test SourceHarmonics __str__ method when nothing is set."""
+    source_harmonics_obj = SourceHarmonics()
+    assert str(source_harmonics_obj) == EXP_STR_NOT_SET
+
+
+def test_source_harmonics___str___all_set(dpf_sound_test_server):
+    """Test SourceHarmonics __str__ method when all data are set."""
+    # Create a field to use in a SourceControlTime object.
+    f_source_control = fields_factory.create_scalar_field(
+        num_entities=1, location=locations.time_freq
+    )
+    f_source_control.append([750, 1250, 3000, 3500], 1)
+    support = TimeFreqSupport()
+    f_time = fields_factory.create_scalar_field(num_entities=1, location=locations.time_freq)
+    f_time.append([0, 1, 2, 3], 1)
+    support.time_frequencies = f_time
+    f_source_control.time_freq_support = support
+
+    # Create a SourceControlTime object.
+    source_control = SourceControlTime()
+    source_control.control = f_source_control
+
+    # Create a SourceHarmonics object.
+    source_harmonics_obj = SourceHarmonics(
+        pytest.data_path_sound_composer_harmonics_source_in_container,
+        source_control,
+    )
+    assert str(source_harmonics_obj) == EXP_STR_ALL_SET
+
+    # Replace source file with one with more less than 30 rpm values and more than 30 orders.
```

Comment:
not a doc comment, but I'm not sure what this means:
_Replace source file with one with more less than 30 rpm values and more than 30 orders._

Reply:
@mattbennett72 Well it means what it says, but probably the phrasing is overly complex. This is here to explain that we do that so that the tests cover all possible code routes (and we treat things differently if there are more or less than X values, for 2 different parameters.
But I agree that it is confusing as it is. I changed it to something simpler.